client/blocks/reader-post-card/byline.jsx

```diff
@@ -0,0 +1,116 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { get, has } from 'lodash';
+
+/**
+ * Internal Dependencies
+ */
+import Gravatar from 'components/gravatar';
+import Gridicon from 'components/gridicon';
+import PostTime from 'reader/post-time';
+import { siteNameFromSiteAndPost } from 'reader/utils';
+import {
+	recordAction,
+	recordGaEvent,
+	recordTrackForPost,
+	recordPermalinkClick
+} from 'reader/stats';
+import ReaderSiteStreamLink from 'blocks/reader-site-stream-link';
+import { getStreamUrl } from 'reader/route';
+import ReaderAuthorLink from 'blocks/reader-author-link';
+
+class PostByline extends React.Component {
+
+	static propTypes = {
+		post: React.PropTypes.object.isRequired,
+		site: React.PropTypes.object,
+		feed: React.PropTypes.object,
+		isDiscoverPost: React.PropTypes.bool
+	}
+
+	static defaultProps = {
+		isDiscoverPost: false
+	}
+
+	recordTagClick = () => {
+		recordAction( 'click_tag' );
+		recordGaEvent( 'Clicked Tag Link' );
+		recordTrackForPost( 'calypso_reader_tag_clicked', this.props.post, {
+			tag: this.props.post.primary_tag.slug
+		} );
+	}
+
+	recordDateClick() {
+		recordPermalinkClick( 'timestamp' );
+		recordGaEvent( 'Clicked Post Permalink', 'timestamp' );
+	}
+
+	render() {
+		const { post, site, feed, isDiscoverPost } = this.props;
+		const feedId = get( feed, 'feed_ID' );
+		const siteId = get( site, 'ID' );
+		const primaryTag = post && post.primary_tag;
+		let siteName = siteNameFromSiteAndPost( site, post );
+		const hasAuthorName = has( post, 'author.name' );
+		const hasMatchingAuthorAndSiteNames = hasAuthorName && siteName.toLowerCase() === post.author.name.toLowerCase();
+		const shouldDisplayAuthor = ! isDiscoverPost && hasAuthorName && ! hasMatchingAuthorAndSiteNames;
+		const streamUrl = getStreamUrl( feedId, siteId );
+
+		if ( ! siteName ) {
+			siteName = this.translate( '(no title)' );
```

Comment:
`this.translate` doesn't exist on react components, only on things from `React.createClass`. I guess this never trips?